### ISSUE TITLE
Cakeday celebration

### ISSUE THREAD

**Issue by jewel-andraia** [honestbleeps/Reddit-Enhancement-Suite#1211]
- Cake badge by username in user menu
- "Happy cakeday! [Click here for a surprise]" notification  (use cakeday badge for background) --> Fireworks!
- a cookie if /u/matheod since he started the convo

**Comment by tao-lu** [comment]
Any interest in resurrecting this issue?

**Comment by jvtrigueros** [comment]
Calculating your cake day is actually very complex!

> The trouble is your cakeday is actually a fairly complex thing. If I remember rightly what /u/deimorz
  said is 
  First take your account creation date stamp and change the year to the current year. Then add 7 days to that. That 7 day period is your cake day eligibility period.
  The first time you have activity (and I'm not sure if that can mean just looking at pages when logged in or if it means voting or posting) that triggers the start of your cake day, which lasts 24 hours.

It was mentioned that if you stick to the "true" cake day meaning that if you just calculate the cake day based on `created_utc`, it should make it a lot easier. So maybe we could start with that?

**Comment by 13steinj** [comment]
I don't believe it has to be calculated-- the cake symbol has it's own unique class so an onclick event can be added, which then just sets a key to the storage saying "user already celebrated his birthday, expire this key in 7 days from Date() whenever checking the user page".
